### ISSUE TITLE
Process does not exit after window is closed

### ISSUE THREAD

**Issue by ataulien** [REGoth-project/REGoth-bs#15]
It seems like bs:f is told to shut down but the process actually seems to stay active until you manually kill it.

**Comment by mworchel** [comment]
On windows I even get an exception when I close the window.
```CPP
void REGothEngine::shutdown()
{
  if (bs::Application::isStarted())
  {
    bs::gDebug().logDebug("[REGothEngine] Shutting down bs::f");

    bs::Application::shutDown();
  }
  else
  {
    bs::gDebug().logWarning("[REGothEngine] Received shutdown request, but bs::f is not running!");
  }
}
```
It can be traced back to the `else` case of the code above. Using the logger after the window (and therefore bs::f) has been shut down seems cause the exception.

**Comment by ataulien** [comment]
Fixed in PR #40
